main.nf

```diff
@@ -92,44 +53,632 @@ if (workflow.profile.contains('awsbatch')) {
+   """
+}
+
+//concatenate all gnomad proteindbs into one
+gnomad_vcf_proteindb
+	.collectFile(name: 'gnomad_proteindb.fa', newLine: false, storeDir: "${baseDir}/result")
+	.set {gnomad_vcf_proteindb_final}
+
+merged_databases = merged_databases.mix(gnomad_vcf_proteindb_final)
+
+/****** cBioPortal mutations *****/
+/**
+ * Download GRCh37 CDS file from ENSEMBL release 75
+ */
+process cds_GRCh37_download{
+
+   when:
+   params.cbioportal
+
+   output:
+   file("Homo_sapiens.GRCh37.75.cds.all.fa") into GRCh37_cds
+
+   script:
+   """
+   wget ftp://ftp.ensembl.org/pub/release-75/fasta/homo_sapiens/cds/Homo_sapiens.GRCh37.75.cds.all.fa.gz
+   gunzip *.gz
+   """
+}
+
+/**
+ * Download all cBioPortal studies using git-lfs
+*/
+ process download_all_cbioportal {
+
+    when:
+         params.cbioportal
+
+   output:
+ 	 file('cbioportal_allstudies_data_mutations_mskcc.txt') into cbio_mutations
+ 	 file('cbioportal_allstudies_data_clinical_sample.txt') into cbio_samples
+
+   script:
+   if (params.cbioportal_study_id == "all")
+        """
+        git clone https://github.com/cBioPortal/datahub.git
+        cd datahub
+        git lfs install --local --skip-smudge
+        git lfs pull -I public --include "data*clinical*sample.txt"
+        git lfs pull -I public --include "data_mutations_mskcc.txt"
+        cd ..
+        cat datahub/public/*/data_mutations_mskcc.txt > cbioportal_allstudies_data_mutations_mskcc.txt
+        cat datahub/public/*/*data*clinical*sample.txt | awk 'BEGIN{FS=OFS="\\t"}{if(\$1!~"#SAMPLE_ID"){gsub("#SAMPLE_ID", "\\nSAMPLE_ID");} print}' | awk 'BEGIN{FS=OFS="\\t"}{s=0; j=0; for(i=1;i<=NF;i++){if(\$i=="CANCER_TYPE_DETAILED") j=1; if(\$i=="CANCER_TYPE") s=1;} if(j==1 && s==0){gsub("CANCER_TYPE_DETAILED", "CANCER_TYPE");} print;}' > cbioportal_allstudies_data_clinical_sample.txt
+        """
+    else
+        """
+        pypgatk_cli.py cbioportal-downloader --config_file "${cbioportal_config}" -d "${params.cbioportal_study_id}"
+        tar -xzvf database_cbioportal/${params.cbioportal_study_id}.tar.gz
+        cat ${params.cbioportal_study_id}/data_mutations_mskcc.txt > cbioportal_allstudies_data_mutations_mskcc.txt
+        cat ${params.cbioportal_study_id}/data_clinical_sample.txt | awk 'BEGIN{FS=OFS="\\t"}{if(\$1!~"#SAMPLE_ID"){gsub("#SAMPLE_ID", "\\nSAMPLE_ID");} print}' | awk 'BEGIN{FS=OFS="\\t"}{s=0; j=0; for(i=1;i<=NF;i++){if(\$i=="CANCER_TYPE_DETAILED") j=1; if(\$i=="CANCER_TYPE") s=1;} if(j==1 && s==0){gsub("CANCER_TYPE_DETAILED", "CANCER_TYPE");} print;}' > cbioportal_allstudies_data_clinical_sample.txt
+        """
+ }
+
+/**
+ * Generate proteinDB from cBioPortal mutations
+ */
+ process cbioportal_proteindb{
+
+   publishDir "${params.outdir}", mode: 'copy', overwrite: true
+
+   when:
+   params.cbioportal
+
+   input:
+   file g from GRCh37_cds
```

Comment:
Channel names should not start with capitals, groovy treats them differently somehow I think.

Reply:
PR #20